### PR TITLE
Add uncuddled else support to the statement_position rule

### PR DESCRIPTION
**Pull request by ateliercw**
implements uncuddled else as an option on statement_position including autocorrect, should solve issue #651

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/RuleConfigurations/StatementPositionConfiguration.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+public enum StatmentModeConfiguration: String {
+    case Standard = "standard", UncuddledElse = "uncuddled_else"
+
+    init(value: AnyObject) throws {
+        guard let string = (value as? String)?.lowercaseString else {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/RuleConfigurations/StatementPositionConfiguration.swift]: this init is equivalent:

```swift
if let string = (value as? String)?.lowercaseString,
       value = StatmentModeConfiguration(rawValue: string) {
    self = value
} else {
    throw ConfigurationError.UnknownConfiguration
}
```

File: Source/SwiftLintFramework/Rules/StatementPositionRule.swift [hunk 2]
```diff
@@ -41,19 +42,85 @@ public struct StatementPositionRule: CorrectableRule, ConfigurationProviderRule
         ]
     )
 
+    public static let uncuddledDescription = RuleDescription(
+        identifier: "statement_position",
+        name: "Statment Position",
+        description: "Else and catch should on the next line, with equal indentation to the " +
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: missing "be":

> Else and catch should be on the next line

File: Source/SwiftLintFramework/Rules/StatementPositionRule.swift [hunk 3]
```diff
@@ -41,19 +42,85 @@ public struct StatementPositionRule: CorrectableRule, ConfigurationProviderRule
...
+        "previous declaration.",
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: could you please indent this string continuation? Just like the previous `description` above.

File: Source/SwiftLintFramework/Rules/StatementPositionRule.swift [hunk 4]
```diff
@@ -41,19 +42,85 @@ public struct StatementPositionRule: CorrectableRule, ConfigurationProviderRule
...
                 location: Location(file: file, characterOffset: range.location))
         }
     }
 
-    public func correctFile(file: File) -> [Correction] {
-        let matches = violationRangesInFile(file, withPattern: pattern)
+    func standardViolationRangesInFile(file: File, withPattern pattern: String) -> [NSRange] {
+        return file.matchPattern(pattern).filter { range, syntaxKinds in
+            return syntaxKinds.startsWith([.Keyword])
+            }.flatMap { $0.0 }
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: please de-indent this one level.

File: Source/SwiftLintFramework/Rules/StatementPositionRule.swift [hunk 5]
```diff
@@ -68,17 +135,117 @@ public struct StatementPositionRule: CorrectableRule, ConfigurationProviderRule
...
+        }
+    }
+
+    static func uncuddledMatchFilter(contents contents: String, syntaxMap: SyntaxMap) ->
+        ((NSTextCheckingResult) -> Bool) {
+        return { match in
+            let range = match.range
+            guard let matchRange = contents.NSRangeToByteRange(start: range.location,
+                                                               length: range.length) else {
+                                                                return false
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/StatementPositionRule.swift]: could you indent this just one level deeper than the `guard`? I find SourceKit's default here to be absolutely bananas.

**Comment by jpsim** [comment]
On the whole this looks great @ateliercw! I just have a few comments. An entry in the changelog would be good too.

**Comment by ateliercw** [comment]
@jpsim adding the the change log entry made me realize that "default" is a more sensible name than "standard", and I made the other formatting and code tweaks that you requested.

**Comment by jpsim** [comment]
Really great first contribution, @ateliercw! 👏

**Pull request merged**